addons/website_event_questions/models/event_question.py

```diff
@@ -25,24 +28,54 @@ def _constrains_event(self):
         if any(question.event_type_id and question.event_id for question in self):
             raise UserError(_('Question cannot belong to both the event category and itself.'))
 
-    @api.model
-    def create(self, vals):
-        event_id = vals.get('event_id', False)
-        if event_id:
-            event = self.env['event.event'].browse([event_id])
-            if event.event_type_id.use_questions and event.event_type_id.question_ids:
-                vals['answer_ids'] = vals.get('answer_ids', []) + [(0, 0, {
-                    'name': answer.name,
-                    'sequence': answer.sequence,
-                }) for answer in event.event_type_id.question_ids.filtered(lambda question: question.title == vals.get('title')).mapped('answer_ids')]
-        return super(EventQuestion, self).create(vals)
-
-
-class EventAnswer(models.Model):
-    _name = 'event.answer'
+    def write(self, vals):
+        """ We add a check to prevent changing the question_type of a question that already has answers.
+        Indeed, it would mess up the event.registration.answer (answer type not matching the question type). """
+
+        if 'question_type' in vals:
```

Comment:
Why not a constraint based on question_type and its answer types ? That way we check that answers are coherent more than writing on answer (not sure it is understandable, i am a bit tired).

Reply:
I tried a "api.constrains" first but I gave up because I could not make it work (something about the _origin stuff that was messing up).
(I don't remember exactly, it was early last week and I have a bad memory sometimes.)
I found that overriding the write was easier in the end.

I could give it another try if it's important / cleaner?